### ISSUE TITLE
classname should be escaped?

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-admin#903]
https://github.com/silverstripe/silverstripe-admin/blob/1/templates/SilverStripe/Admin/Includes/ModelAdmin_Content.ss#L19

Have not researched this in great depth, but when I do a search for $ClassName in *.ss files, this module pops up as using them a few times.

I dont think backslashes are allowed in css classes, so this may need to be changed?

**Comment by robbieaverill** [comment]
I think changing this would break things for people. We've got implementations in supported modules that depend on the class name being a PHP FCQN in JavaScript and in CSS

**Comment by sunnysideup** [comment]
change it in SS5?

**Comment by robbieaverill** [comment]
Yeah that sounds better. To be honest, I don't think the PHP class name belongs in the frontend code at all... perhaps we can develop a better API for representing backend models in frontend templates and code...

**Comment by sunnysideup** [comment]
We should add a new method to Text / Varchar called: HTMLClassName / SHortClassName (if there is not one already), that outputs the className like this: Silverstripe-Hello-World-Class in the way ModelAdmin uses it.  Perhaps this method could even be added to DataObject (although I am sure ya'll are keen to reduce the size of this class).

**Comment by michalkleiner** [comment]
There already is a method to obtain it via `ClassInfo`. We currently use an extensions applied to `Controller` and `DataObject` having a simple method like

```
public function getClassShortName()
{
    return ClassInfo::shortName($this->owner);
}
```

**Comment by michalkleiner** [comment]
I guess it can be made available via `TemplateGlobalProvider` on ViewableData.

**Comment by sunnysideup** [comment]
@michalkleiner , we do that too, but I think a standard, universal "HTML ClassName" would ensure that more devs would use that standardised one, allowing modules to connect more seamlessly, etc...

**Comment by michalkleiner** [comment]
Yep, consistent "core" way would be good, I agree with that.

Regardless of how we arrive there, the short name might not be what we need though, we used it to retrofit some features after SS3 upgrade, but it may, in rare cases, not always represent the class one wants to target in SS4.

So something like replacing `\` with `-` or `_` and lowercasing might be the way. Just thinking out loud.
